c_api/docs/taichi/taichi_core.h.md

```diff
@@ -0,0 +1,410 @@
+
+For convenience, in the following text (and other C-API documentations), the term **host** refers to the user of the C-API; the term **device** refers to the logical (conceptual) compute device that Taichi Runtime offloads its compute tasks to. A *device* might not be an actual discrete processor away from the CPU and the *host* MAY NOT be able to access the memory allocated on the *device*.
+
+Unless explicitly explained, **device**, **backend**, **offload targer** and **GPU** are used interchangeably; **host**, **user code**, **user procedure** and **CPU** are used interchangeably too.
+
+## How to...
+
+In this section we give an brief introduction about what you might want to do with the Taichi C-API.
+
+### Create and destroy a Runtime Instance
+
+To work with Taichi, you first create an runtime instance. You SHOULD only create a single runtime per thread. Currently we don't officially claim that multiple runtime instances can coexist in a process, please feel free to [report issues](https://github.com/taichi-dev/taichi/issues) if you encountered any problem with such usage.
+
+```cpp
+TiRuntime runtime = ti_create_runtime(TI_ARCH_VULKAN);
+```
+
+When your program reaches the end, you SHOULD destroy the runtime instance. Please ensure any other related resources have been destroyed before the `handle.runtime` itself.
+
+```cpp
+ti_destroy_runtime(runtime);
+```
+
+### Allocate and Free Memory
+
+Allocate a piece of memory that is only visible to the device. On GPU backends, it usually means that the memory is located in the graphics memory (GRAM).
+
+```cpp
+TiMemoryAllocateInfo mai {};
+mai.size = 1024; // Size in bytes.
+mai.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+TiMemory memory = ti_allocate_memory(runtime, &mai);
+```
+
+You MAY free allocated memory explicitly; but memory allocations will be automatically freed when the related `handle.runtime` is destroyed.
+
+```cpp
+ti_free_memory(runtime, memory);
+```
+
+### Allocate Host-Accessible Memory
+
+By default, memory allocations are physically or conceptually local to the offload target for performance reasons. You can configure the allocate info to enable host access to memory allocations. But please note that host-accessible allocations MAY slow down computation on GPU because of the limited bus bandwidth between the host memory and the device.
+
+To allow data to be streamed into the memory, `host_write` MUST be set true.
+
+```cpp
+TiMemoryAllocateInfo mai {};
+mai.size = 1024; // Size in bytes.
+mai.host_write = true;
+mai.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+TiMemory steaming_memory = ti_allocate_memory(runtime, &mai);
+
+// ...
+
+std::vector<uint8_t> src = some_random_data_source();
+
+void* dst = ti_map_memory(runtime, steaming_memory);
+std::memcpy(dst, src.data(), src.size());
+ti_unmap_memory(runtime, streaming_memory);
+```
+
+To read data back to the host, `host_read` MUST be set true.
```

Comment:
The word "read" does not sound about right to me... "Write"? Maybe it's just that I'm new to it.